core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MemoryManagement.kt

```diff
@@ -8,12 +8,11 @@ import io.ktor.util.date.*
 internal object MemoryManagement {
 
   internal suspend fun addMemoriesAfterStream(
-    chat: Chat,
-    request: ChatCompletionRequest,
+    chat: LLM,
+    lastRequestMessage: Message?,
```

Comment:
For me, it's weird when we are sending a nullable object and we don't do anything inside if the object is null

Maybe our method should be

```kotlin
internal suspend fun addMemoriesAfterStream(
    chat: LLM,
    message: Message,
    scope: Conversation,
    messages: List<Message>,
  )
```

and then

```kotlin
request.messages.lastOrNull()?.let {
          MemoryManagement.addMemoriesAfterStream(
            this@Chat,
            it,
            scope,
            listOf(message)
          )
        }
```

Reply:
Adressed in #440